src/user-pool.ts

```diff
@@ -0,0 +1,233 @@
+  /**
+   * Verifies the authentication challenge response.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-verify-auth-challenge-response.html
+   * @default - no trigger configured
+   */
+  readonly verifyAuthChallengeResponse?: Trigger<lambda.VerifyAuthChallengeResponseTriggerEvent>;
+  /**
+   * Amazon Cognito invokes this trigger to send email notifications to users.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-email-sender.html
+   * @default - no trigger configured
+   */
+  readonly customEmailSender?: Trigger<lambda.CustomEmailSenderTriggerEvent>;
+  /**
+   * Amazon Cognito invokes this trigger to send SMS notifications to users.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-custom-sms-sender.html
+   * @default - no trigger configured
+   */
+  readonly customSmsSender?: Trigger<lambda.CustomSMSSenderTriggerEvent>;
+}
+
+// map a friendly trigger name to the {@link aws_cognito.UserPoolOperation}.
+const triggers: {
+  [triggerName in keyof UserPoolTriggers]: aws_cognito.UserPoolOperation;
+} = {
+  createAuthChallenge: aws_cognito.UserPoolOperation.CREATE_AUTH_CHALLENGE,
+  customEmailSender: aws_cognito.UserPoolOperation.CUSTOM_EMAIL_SENDER,
+  customMessage: aws_cognito.UserPoolOperation.CUSTOM_MESSAGE,
+  customSmsSender: aws_cognito.UserPoolOperation.CUSTOM_SMS_SENDER,
+  defineAuthChallenge: aws_cognito.UserPoolOperation.DEFINE_AUTH_CHALLENGE,
+  postAuthentication: aws_cognito.UserPoolOperation.POST_AUTHENTICATION,
+  postConfirmation: aws_cognito.UserPoolOperation.POST_CONFIRMATION,
+  preAuthentication: aws_cognito.UserPoolOperation.PRE_AUTHENTICATION,
+  preSignUp: aws_cognito.UserPoolOperation.PRE_SIGN_UP,
+  preTokenGeneration: aws_cognito.UserPoolOperation.PRE_TOKEN_GENERATION,
+  userMigration: aws_cognito.UserPoolOperation.USER_MIGRATION,
+  verifyAuthChallengeResponse:
+    aws_cognito.UserPoolOperation.VERIFY_AUTH_CHALLENGE_RESPONSE,
+};
+
+export interface UserPoolProps
+  extends Omit<aws_cognito.UserPoolProps, "lambdaTriggers"> {
+  /**
+   * Lambda functions to use for supported Cognito triggers.
+   * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html
+   * @default - No Lambda triggers.
+   */
+  readonly lambdaTriggers?: UserPoolTriggers;
+}
+
+/**
+ * Defines an AWS Cognito User Pool.
+ */
+export class UserPool {
+  public static readonly FunctionlessType = "UserPool";
+  public readonly FunctionlessType = UserPool.FunctionlessType;
+
+  /**
+   * The underlying {@link aws_cognito.UserPool} instance.
+   */
+  readonly resource: aws_cognito.UserPool;
+
+  /**
+   * Create a new {@link UserPool} from an existing {@link aws_cognito.UserPool}.
+   *
+   *
+   * @param resource an existing UserPool Construct.
+   * @see {@link aws_cognito.UserPool}
+   */
+  constructor(resource: aws_cognito.UserPool);
```

Comment:
All of the import cases should use a "from" method. #134

Reply:
I can add the `from` method but am I forced to remove the constructor? I don't need a `BaseUserPool`.

Reply:
I guess we have been removing them, shouldn't we be consistent? The harm would be that multiple constructor overrides would get messy or if there are non-import only methods, it would be harder to model that.

For example StepFunctions exposes `definition`, unless its an imported Sfn. 

Maybe its more of an issue for transpiled resources.

The argument for keeping the constructor is to avoid the base class and inheritance?

Reply:
Mostly to avoid an extra class but I am fine with changing it to be consistent. I usually hate Base* classes but I also get the point off it confusing developers. The first constructor they see is the import one so that could be a problem. I can roll in a fix for secrets into this change.